iothub/device/src/InternalClient.ConventionBasedOperations.cs

```diff
@@ -32,5 +33,33 @@ internal Task SendTelemetryAsync(TelemetryMessage telemetryMessage, Cancellation
 
             return SendEventAsync(telemetryMessage, cancellationToken);
         }
+
+        internal Task SubscribeToCommandsAsync(Func<CommandRequest, object, Task<CommandResponse>> callback, object userContext, CancellationToken cancellationToken)
+        {
+            // Subscribe to methods default handler internally and use the callback received internally to invoke the user supplied command callback.
+            var methodDefaultCallback = new MethodCallback(async (methodRequest, userContext) =>
+            {
+                CommandRequest commandRequest;
+                if (methodRequest.Name.Contains(ConventionBasedConstants.ComponentLevelCommandSeparator))
```

Comment:
any chance methodRequest or methodRequest.Name can be null?

Reply:
there won't be, a method request will always be accompanied by method name; but no harm in adding a null check here